open-craft/openedx-ltistore
# feat: LTI 1.3 reusable configuration

## Description

This PR adds compatibility for LTI 1.3 external configuration to the openedx-ltistore. This change adds missing fields required for LTI 1.3 external reusable configurations to work.

## Type of Change

- [x] Add LTI 1.3 fields to ExternalLtiConfiguration model.
- [x] Include unit tests for all added and modified code.

## Testing

1. Install xblock-lti-consumer (Use the related PR: https://github.com/openedx/xblock-lti-consumer/pull/390)
2. Install and configure openedx-ltistore (follow the setup steps from: https://github.com/openedx/xblock-lti-consumer/pull/239 and use the PR related to this ticket)
3. Create an IMS reference tool: https://lti-ri.imsglobal.org/lti/tools
4. Go to http://localhost:18000/admin/lti_store/externallticonfiguration/ and create a new external LTI 1.3 configuration.
5. Fill the "LTI 1.3 Private Key",  "LTI 1.3 Client ID" and "LTI 1.3 Tool Keyset URL" or "LTI 1.3 Tool Public Key" field.
6. Create a course and add a LTI 1.3 consumer XBlock.
7. Set the consumer XBlock to user an external configuration and set the reusable configuration ID (follow the steps on how to get the filter key from: https://github.com/openedx/xblock-lti-consumer/pull/239) 
8. Copy the keyset and access token URL from the XBlock preview back to the IMS reference tool previously created. 
9. Go to the course a launch the XBlock.
10. The LTI 1.3 launch should be valid.
